File: Software/Python/Examples/Grove_LED.py
#!/usr/bin/env python
#
# https://www.dexterindustries.com/GoPiGo/
# https://github.com/DexterInd/GoPiGo3
#
# Copyright (c) 2017 Dexter Industries
# Released under the MIT license (http://choosealicense.com/licenses/mit/).
# For more information see https://github.com/DexterInd/GoPiGo3/blob/master/LICENSE.md
#
# This code is an example for using a grove LED with GoPiGo3.
#
# Hardware: Connect a grove LED to port AD2 of the GoPiGo3.
#
# Results: When you run this program, the grove LED should turn on for two seconds.

from __future__ import print_function # use python 3 syntax but make it compatible with python 2
from __future__ import division       #                           ''

import time     # import the time library for the sleep function
import gopigo3 # import the GoPiGo3 drivers

GPG = gopigo3.GoPiGo3() # Create an instance of the GoPiGo3 class. GPG will be the GoPiGo3 object.

try:
    GPG.set_grove_type(GPG.GROVE_2, GPG.GROVE_TYPE.CUSTOM)
    GPG.set_grove_mode(GPG.GROVE_2_1, GPG.GROVE_OUTPUT_DIGITAL)
    GPG.set_grove_state(GPG.GROVE_2_1, True)
    #GPG.set_grove_mode(GPG.GROVE_2_1, GPG.GROVE_OUTPUT_PWM)
    #GPG.set_grove_pwm_duty(GPG.GROVE_2_1, 50)
    
    time.sleep(2)
    GPG.reset_all()

except KeyboardInterrupt: # except the program gets interrupted by Ctrl+C on the keyboard.
    GPG.reset_all()        # Unconfigure the sensors, disable the motors, and restore the LED to the control of the GoPiGo3 firmware.

File: Software/C/Examples/grove_led.cpp
/*
 *  https://www.dexterindustries.com/GoPiGo3/
 *  https://github.com/DexterInd/GoPiGo3
 *
 *  Copyright (c) 2017 Dexter Industries
 *  Released under the MIT license (http://choosealicense.com/licenses/mit/).
 *  For more information see https://github.com/DexterInd/GoPiGo3/blob/master/LICENSE.md
 *
 *  This is an example for using an ultrasonic sensor and IR receiver with the GoPiGo3.
 *
 *  Hardware: Connect a grove LED to GPG3 AD1 port.
 *
 *  Results:  When you run this program, you should see the grove LED brightness pulsing up and down.
 *
 *  Example compile command:
 *    g++ -o program "grove_led.cpp"
 *  Example run command:
 *    sudo ./program
 *
 */

#include "GoPiGo3.cpp" // for GoPiGo3
#include <stdio.h>      // for printf
#include <unistd.h>     // for usleep
#include <signal.h>     // for catching exit signals

GoPiGo3 GPG;

void exit_signal_handler(int signo);

int main(){
  signal(SIGINT, exit_signal_handler); // register the exit function for Ctrl+C

  GPG.detect(); // Make sure that the GoPiGo3 is communicating and that the firmware is compatible with the drivers.

  GPG.set_grove_type(GROVE_1, GROVE_TYPE_CUSTOM);
  GPG.set_grove_mode(GROVE_1_1, OUTPUT_PWM);

  int8_t i = 0;
  int8_t a = 1;
  while(true){
    GPG.set_grove_pwm_duty(GROVE_1_1, i);
    i += a;
    if(i == 100 || i == 0){
      a *= -1;
    }
    usleep(5000);
  }
}

// Signal handler that will be called when Ctrl+C is pressed to stop the program
void exit_signal_handler(int signo){
  if(signo == SIGINT){
    GPG.reset_all();    // Reset everything so there are no run-away motors
    exit(-2);
  }
}
